dao/dao.go

```diff
@@ -11,7 +11,10 @@ type DAO interface {
 	GetPaymentResource(string) (*models.PaymentResourceDB, error)
 	PatchPaymentResource(id string, paymentUpdate *models.PaymentResourceDB) error
 	GetPaymentResourceByProviderID(providerID string) (*models.PaymentResourceDB, error)
-	CreateBulkRefund(providerID string, bulkRefund models.BulkRefundDB) error
+	GetPaymentResourceByExternalPaymentTransactionID(providerID string) (*models.PaymentResourceDB, error)
+	CreateBulkRefund(providerID string, bulkRefund models.BulkRefundDB, idQuery string) error
```

Comment:
This isn't called anywhere so can be removed. You should still be able to call `func (m *MongoService) CreateBulkRefund....` from within the mongo.go file as you are currently without having it in this interface